src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.ts

```diff
@@ -57,12 +75,26 @@ export class ExamChecklistComponent implements OnChanges, OnInit, OnDestroy {
         this.allGroupsContainExercise = this.examChecklistService.checkEachGroupContainsExercise(this.exam);
         this.countMandatoryExercises = this.exam.exerciseGroups?.filter((group) => group.isMandatory)?.length ?? 0;
         this.hasOptionalExercises = this.countMandatoryExercises < (this.exam.exerciseGroups?.length ?? 0);
+        if (this.exam?.id) {
```

Comment:
is it really necessary to check this on each change? Could we not just check this once in NgOnInit?

Reply:
Changed